core/foundation/inc/ROOT/RLogger.hxx

```diff
@@ -176,13 +317,27 @@ public:
 #define R__LOG_PRETTY_FUNCTION __PRETTY_FUNCTION__
 #endif
 
-#define R__LOG_HERE(LEVEL, GROUP) \
-   ROOT::Experimental::RLogEntry(LEVEL, GROUP).SetFile(__FILE__).SetLine(__LINE__).SetFunction(R__LOG_PRETTY_FUNCTION)
-
-#define R__FATAL_HERE(GROUP) R__LOG_HERE(ROOT::Experimental::ELogLevel::kFatal, GROUP)
-#define R__ERROR_HERE(GROUP) R__LOG_HERE(ROOT::Experimental::ELogLevel::kError, GROUP)
-#define R__WARNING_HERE(GROUP) R__LOG_HERE(ROOT::Experimental::ELogLevel::kWarning, GROUP)
-#define R__INFO_HERE(GROUP) R__LOG_HERE(ROOT::Experimental::ELogLevel::kInfo, GROUP)
-#define R__DEBUG_HERE(GROUP) R__LOG_HERE(ROOT::Experimental::ELogLevel::kDebug, GROUP)
+// use `&&` to prevent "ambiguous else" outside macro: `if (...) R__DEBUG_HERE()...`
+#define R__LOG_HERE(LEVEL, CHANNEL) \
+   ROOT::Experimental::Internal::ShouldCreateLogEntry(LEVEL, CHANNEL) && \
+     ROOT::Experimental::RLogEntry(LEVEL, CHANNEL).SetFile(__FILE__).SetLine(__LINE__).SetFunction(R__LOG_PRETTY_FUNCTION)
+
+///\{
+/// \name Macros to create an RLogEntry
+#define R__FATAL_HERE(CHANNEL) R__LOG_HERE(ROOT::Experimental::ELogLevel::kFatal, CHANNEL)
+#define R__ERROR_HERE(CHANNEL) R__LOG_HERE(ROOT::Experimental::ELogLevel::kError, CHANNEL)
+#define R__WARNING_HERE(CHANNEL) R__LOG_HERE(ROOT::Experimental::ELogLevel::kWarning, CHANNEL)
+#define R__INFO_HERE(CHANNEL) R__LOG_HERE(ROOT::Experimental::ELogLevel::kInfo, CHANNEL)
+#define R__DEBUG_HERE(CHANNEL) R__LOG_HERE(ROOT::Experimental::ELogLevel::kDebug, CHANNEL)
+///\}
+
+///\{
+/// \name Macros to create an RLogEntry with specific severity levels auch as `kError + 5`
+#define R__NFATAL_HERE(CHANNEL, LEVEL) R__LOG_HERE(ROOT::Experimental::ELogLevel::kFatal + LEVEL, CHANNEL)
+#define R__NERROR_HERE(CHANNEL, LEVEL) R__LOG_HERE(ROOT::Experimental::ELogLevel::kError + LEVEL, CHANNEL)
+#define R__NWARNING_HERE(CHANNEL, LEVEL) R__LOG_HERE(ROOT::Experimental::ELogLevel::kWarning + LEVEL, CHANNEL)
+#define R__NINFO_HERE(CHANNEL, LEVEL) R__LOG_HERE(ROOT::Experimental::ELogLevel::kInfo + LEVEL, CHANNEL)
+#define R__NDEBUG_HERE(CHANNEL, LEVEL) R__LOG_HERE(ROOT::Experimental::ELogLevel::kDebug + LEVEL, CHANNEL)
```

Comment:
maybe `OFFSET` rather than `LEVEL`?

Reply:
`SEVERITY`? ("Offset" to what, and it just says it's a difference but doesn't come with a meaning / "unit")

Reply:
sure, it was just `LEVEL` that was overloaded